File: payment/src/main/java/com/myStore/payment/model/Payment.java
package com.myStore.payment.model;

import com.myStore.payment.model.enums.PaymentStatusEnum;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "payments")
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@Builder
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "order_id", nullable = false)
    private UUID orderId;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "provider", nullable = false, length = 50)
    @Builder.Default
    private String provider = "MERCADO_PAGO";

    @Column(name = "transaction_id", length = 255)
    private String transactionId;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 30)
    @Builder.Default
    private PaymentStatusEnum status = PaymentStatusEnum.PENDING;

    @Column(name = "amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(name = "init_point", columnDefinition = "TEXT")
    private String initPoint;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
}

File: frontend/src/domain/models/Payment.ts
import { z } from 'zod';

export const PaymentStatusEnumSchema = z.enum(['PENDING', 'APPROVED', 'REJECTED', 'CANCELLED']);
export type PaymentStatus = z.infer<typeof PaymentStatusEnumSchema>;

export const PaymentSchema = z.object({
  id: z.string(),
  orderId: z.string(),
  userId: z.string(),
  provider: z.string(),
  transactionId: z.string().nullable().optional(),
  status: PaymentStatusEnumSchema,
  amount: z.number(),
  initPoint: z.string().nullable().optional(),
  createdAt: z.string(),
});

export type Payment = z.infer<typeof PaymentSchema>;

// Alias for backwards compatibility
export type PaymentDetails = Payment;
